### ISSUE TITLE
Peacock 2 Preferred Block Ordering

### ISSUE THREAD

**Issue by friedmud** [idaholab/moose#10473]
## Rationale
We like our input files to be ordered in a particular ordering... for instance:

Mesh
Variables
ICs
Kernels
BCs
Executioner
Postprocessors
Output

The original Peacock tried to preserve this "preferred" ordering when possible... but also kept the existing ordering in an input file when provided.

This was also true for parameters as well.

## Description
The original capability worked by providing an example input file that showed the "preferred" ordering to Peacock and then Peacock would attempt do its best to meet that ordering.  You can see the example input file here: https://github.com/idaholab/moose/blob/devel/gui/input_template

In addition, each application could provide its own `input_template` to override this ordering.

The code that did the sorting is here:

Template file reading:
https://github.com/idaholab/moose/blob/devel/gui/gui/InputFileWidget.py#L46-L55

Template and existing file merging:
https://github.com/idaholab/moose/blob/devel/gui/gui/InputFileTextbox.py#L59-L68

Merging code:
https://github.com/idaholab/moose/blob/devel/gui/utils/mergeLists.py#L5

## Impact
Better input files from Peacock

**Comment by friedmud** [comment]
A little more on this:

The idea is to create a sorted list of sections / parameter names... then you loop over _that_ and check to see if you have those in your current tree and output them if you do.

**Comment by friedmud** [comment]
@brianmoose any movement on this?  This is really needed.

**Comment by brianmoose** [comment]
Not yet, I will start on it.

**Comment by brianmoose** [comment]
What I got from the discussion yesterday was that all peacock has to do is write out input files using the hit parser. The hit parser will format to a specific standard and take care of this preferred ordering.
@friedmud  Correct?

**Comment by friedmud** [comment]
Yep - in the meantime you might want to make Peacock use the HIT parser for
outputting files so it's ready to go when that capability lands.

Derek

**Comment by rwcarlsen** [comment]
I did add a change to the peacock InputFileWriter to use the formatter - I'm not sure if there are any other spots that need to be updated though - @brianmoose?

**Comment by rwcarlsen** [comment]
referring to #12078 that is

**Comment by brianmoose** [comment]
Yeah, I think `InputTreeWriter` should be the only place
